File: 2023/Leetcode/problem1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        complements = {}
        
        for i in range(len(nums)):
            if nums[i] not in complements:
                complements[target - nums[i]] = i
            else:
                return [complements[nums[i]], i]

File: 2022/Leetcode/problem1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> result(2);
        unordered_map<int, int> goalIndexMap;
        
        for (int i = 0; i < nums.size(); i++) {
            if (goalIndexMap.find(nums[i]) == goalIndexMap.end()) {
                goalIndexMap[target - nums[i]] = i;
            } else {
                result[0] = i;
                result[1] = goalIndexMap[nums[i]];
                break;
            }
        }
        return result;
    }
};